main.cpp

```diff
@@ -76,9 +76,656 @@ Purpose:  This project continues developing Project3.
+
+            while(numItemsAdded > 0 && itemsToRemove > 0)
+            {
+                numItemsAdded--;
+                itemsToRemove--;
+            }
+            std::cout << numItemsAdded << " items remaining" << std::endl;
+        }
+        else
+            std::cout << "Pencilcase is closed or nothing inside" << std::endl;
+    }
+    else
+        std::cout << "Can't remove " << itemsToRemove << " items"  << std::endl;
+}
+
+// ============= Drill Functions =================
+
+Drill::Drill() : isPlugged(true), drillbitAttached(false), powerInWatts(3600) 
+{
+    std::cout << "****** Drill Created ******" << std::endl;
+}
+
+Drill::~Drill()
+{
+    std::cout << "****** Drill Destroyed ******" << std::endl;
+}
+
+void Drill::drillHole(int holes)
+{
+    if(isPlugged && drillbitAttached && holes > 0)
+    {
+        std::cout << "Drilling holes..." << std::endl;
+        std::cout << "Drilling done!!" << std::endl;
+    }
+    else if(holes < 1)
+        std::cout << "You can't drill " << holes << " holes..." << std::endl;
+    else 
+        std::cout << "Whoops! Check if your drill is ready" << std::endl;
+}
+
+void Drill::unplug()
+{
+    isPlugged = false;
+    std::cout << "Unplugged drill" << std::endl;
+}
+
+void Drill::attachDrillbit(float drillbitWidth)
+{
+    rotationsToSecureBit = 0.8f * drillbitWidth;
+
+    std::cout << "Securing drill bit... " << std::endl;
+
+    for(int i = 0; i < int(rotationsToSecureBit); ++i)
+        chuckTightness+= rotationsToSecureBit;
+
+    drillbitAttached = true;	
+
+    std::cout << "Drillbit secured" << std::endl;
+}
+
+// =========== ElevatorUser Functions ===========
+
+Elevator::ElevatorUser::ElevatorUser() : weight(100.0), destinationFloor(0), currentFloor(0)
+{
+    std::cout << "****** ElevatorUser Created ******" << std::endl;
+}
+
+Elevator::ElevatorUser::~ElevatorUser()
+{
+    std::cout << "****** ElevatorUser Destroyed ******" << std::endl;
+}
+
+void Elevator::ElevatorUser::callElevator()
+{
+    if(!isInsideElevator)
+        std::cout << "Calling elevator..." << std::endl;
+    else
+        std::cout << "Wait till you exit the elevator" << std::endl;
+}
+
+void Elevator::ElevatorUser::enterElevator()
+{
+    if(!isInsideElevator)
+    {
+        isInsideElevator = true;
+        std::cout << "You have entered the elevator, now go to a floor" << std::endl;
+    }
+    else
+        std::cout << "You are already inside the elevator" << std::endl;
+}
+
+void Elevator::ElevatorUser::setDestination(int newDestinationFloor)
+{
+    destinationFloor = newDestinationFloor;
+    std::cout << "Destination floor " << destinationFloor << std::endl;
+}
+
+void Elevator::ElevatorUser::exitElevator()
+{
+    std::cout << "Exiting... ";
+
+    if(isInsideElevator)
+    {
+        isInsideElevator = false;
+        std::cout << "You have exited the elevator" << std::endl;
+    }    
+    else
+        std::cout << "You are not in an elevator yet!" << std::endl;
+}
```

Comment:
If you use curly braces for the `if()` block, use them for the `else if` and `else` blocks too.
- [ ] revise